Xamarin.PropertyEditing.Mac/Controls/StringEditorControl.cs

```diff
@@ -75,5 +75,13 @@ protected override void UpdateAccessibilityValues ()
 		{
 			StringEditor.AccessibilityTitle = string.Format (LocalizationResources.AccessibilityString, ViewModel.Property.Name);
 		}
+
+		protected override void OnPropertyChanged (object sender, System.ComponentModel.PropertyChangedEventArgs e)
+		{
+			base.OnPropertyChanged (sender, e);
+
+			if (e.PropertyName == "ValueSource")
+				PropertyButton.ValueSourceChanged (ViewModel.ValueSource);
```

Comment:
We shouldn't have to hook this up for every editor. Ideally we'd pass the view model to the property button and it'd hook everything up. Barring that at least do it where you've hooked up the reset command.

Reply:
It was the only editor hooked up this way as I'd overlooked a required refactor. Should be fixed in my next push.